src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -1,15 +1,70 @@
 package wolox.training.controllers;
 
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.DeleteMapping;
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import wolox.training.exceptions.BookIdMismatchException;
+import wolox.training.exceptions.BookNotFoundException;
+import wolox.training.models.Book;
+import wolox.training.repositories.BookRepository;
 
 @Controller
 public class BookController {
+    @Autowired
+    private BookRepository bookRepository;
```

Comment:
you can move all the logic that is bind with the repository to a service. It's a good practice to make a class "service" that calles the repository(in this case "BookService"); and from the repository you would called this service, not directly the repo. Check this post https://medium.com/karibu-blog/c%C3%B3mo-aplicar-jpa-en-spring-boot-e45ac5bee3b4